src/crimson/osd/osd.cc

```diff
@@ -276,21 +276,6 @@ seastar::future<> OSD::_send_boot()
   return monc->send_message(m);
 }
 
```

Comment:
msgr shutdown can't be in stop function since the stop fun sequence can't be guaranteed when engine_exit() , so move it to osd::shutdown(), explicitly call shutdown() before engine_exit();  crimson-osd --mkfs will exit, so stop() fun will be executed, so can't put shutdown() in stop() for mkfs since these msgrs are not started for mkfs.

Reply:
yeah, i think we need to promote the sharded service to global services instead of pretending that they are member variables in `OSD` or `Heartbeat`. see https://github.com/ceph/ceph/projects/2#card-18842255